src/anomalib/data/utils/augmenter.py

```diff
@@ -168,3 +170,75 @@ def augment_batch(self, batch: Tensor) -> tuple[Tensor, Tensor]:
         augmented_batch = batch * (1 - masks) + (beta) * perturbations + (1 - beta) * batch * (masks)
 
         return augmented_batch, masks
+
+
+class PerlinROIAugmenter(Augmenter):
+    def augment_batch(self, batch: Tensor, thresh_batch: Tensor | None = None) -> tuple[Tensor, Tensor]:
+        """Generate anomalous augmentations for a batch of input images.
+
+        Args:
+            batch (Tensor): Batch of input images
+            thresh_batch (Tensor | None, optional): Batch of thresholds
+
+        Returns:
+            - Augmented image to which anomalous perturbations have been added.
+            - Ground truth masks corresponding to the anomalous perturbations.
+        """
+        batch_size, channels, height, width = batch.shape
+
+        # Collect perturbations
+        perturbations_list = []
+        masks_list = []
+        for _ in range(batch_size):
+            if torch.rand(1) > self.p_anomalous:  # include normal samples
+                perturbations_list.append(torch.zeros((channels, height, width)))
+                masks_list.append(torch.zeros((1, height, width)))
+            else:
+                anomaly_source_path = (
+                    random.sample(self.anomaly_source_paths, 1)[0] if len(self.anomaly_source_paths) > 0 else None
+                )
+                perturbation, mask = self.generate_perturbation(height, width, anomaly_source_path)
+                perturbations_list.append(Tensor(perturbation).permute((2, 0, 1)))
+                masks_list.append((Tensor(mask).permute((2, 0, 1)) * thresh_batch))
+
+        perturbations = torch.stack(perturbations_list).to(batch.device)
+        masks = torch.stack(masks_list).to(batch.device)
+
+        # Apply perturbations batch wise
+        if isinstance(self.beta, float):
+            beta = self.beta
+        elif isinstance(self.beta, tuple):
+            beta = torch.rand(batch_size) * (self.beta[1] - self.beta[0]) + self.beta[0]
+            beta = beta.view(batch_size, 1, 1, 1).expand_as(batch).to(batch.device)  # type: ignore
+        else:
+            raise ValueError("Beta must be either float or tuple of floats")
+
+        augmented_batch = batch * (1 - masks) + (beta) * (perturbations * masks) + (1 - beta) * batch * (masks)
+
+        return augmented_batch, masks
+
+    def gaussian_blur(self, image: Tensor, transform: A.Compose) -> tuple[Tensor, Tensor]:
+        """Generate Gaussian Blur.
+
```

Comment:
Can you also describe the intention behind performing gaussian blur and finding out the thresholds based on it.

Reply:
we renamed the function `generate_roi` as it does more than gaussian blur. The intention is to generate the region of interest.